### PR TITLE
refactor(tools): update references from kong.tools.utils to kong.tools.table

### PR DESCRIPTION
**Pull request by Water-Melon**
KAG-3138

### CODE REVIEW

File: kong/pdk/log.lua
Comment (chronolaw) [inline kong/pdk/log.lua:38]: should be require("kong.tools.string").split

File: kong/tracing/instrumentation.lua
Comment (chronolaw) [inline kong/tracing/instrumentation.lua:20]: should be require("kong.tools.time").time_ns

File: spec/helpers/http_mock/nginx_instance.lua
Comment (Water-Melon) [inline spec/helpers/http_mock/nginx_instance.lua:17]: If this is changed to `kong.tools.table`, the upgrade test will fail. Seems that there is no `kong/tools/table.lua` file in 2.8.0 codebase.

Comment (chronolaw) [reply spec/helpers/http_mock/nginx_instance.lua:17]: please add a comment for this.

File: spec/03-plugins/15-response-transformer/01-header_transformer_spec.lua
Comment (Water-Melon) [inline spec/03-plugins/15-response-transformer/01-header_transformer_spec.lua:28]: If only the subsystem is present, `jit-uuid` will throw an error due to the absence of `nginx_configure`.

File: spec/03-plugins/15-response-transformer/01-header_transformer_spec.lua
Comment (Water-Melon) [inline spec/03-plugins/15-response-transformer/01-header_transformer_spec.lua:32]: If re is not set, `jit-uuid` will throw an error because it can't find `ngx.re.find`.

File: spec/helpers/http_mock/nginx_instance.lua
Comment (Water-Melon) [reply spec/helpers/http_mock/nginx_instance.lua:19]: Added

File: Makefile
Comment (Water-Melon) [inline Makefile:4]: Force the use of the luasocket module's unix feature to avoid this issue: https://github.com/Kong/kong/actions/runs/9014154644/job/24766292897?pr=12487

**Comment by Water-Melon** [comment]
full matrix tests: https://github.com/Kong/kong/actions/runs/9056770720/job/24879982725?pr=12487

**Comment by Water-Melon** [comment]
Please merge this PR after merging https://github.com/Kong/kong/pull/13026

**Comment by Water-Melon** [comment]
CI tests after removing most exported functions of `kong\tools\table.lua`
https://github.com/Kong/kong/pull/12487/files#diff-288ad830a31e2aee02e9bdb68009d2ee3daab8e4ff39247113e598a02a7640f6R20-R37

https://github.com/Kong/kong/actions/runs/9072658583/job/24928507132?pr=12487

File: kong/db/dao/targets.lua
Comment (chronolaw) [inline kong/db/dao/targets.lua:14]: Could be `local table_merge = require("kong.tools.table").table_merge`?

File: kong/db/schema/others/declarative_config.lua
Comment (chronolaw) [inline kong/db/schema/others/declarative_config.lua:2]: This line may be conflict with #12952, could we keep it?

File: kong/db/strategies/postgres/connector.lua
Comment (chronolaw) [inline kong/db/strategies/postgres/connector.lua:34]: Could be `local table_merge = require("kong.tools.table").table_merge`?

File: kong/plugins/opentelemetry/otlp.lua
Comment (chronolaw) [inline kong/plugins/opentelemetry/otlp.lua:7]: should we remove this line? And replace it by `local table_merge = require("kong.tools.table").table_merge `

File: spec/01-unit/09-balancer/05-worker_consistency_spec.lua
Comment (chronolaw) [inline spec/01-unit/09-balancer/05-worker_consistency_spec.lua:5]: It may conflict with #12952

File: kong/db/dao/targets.lua
Comment (Water-Melon) [reply kong/db/dao/targets.lua]: updated

File: kong/db/strategies/postgres/connector.lua
Comment (Water-Melon) [reply kong/db/strategies/postgres/connector.lua]: changed

File: kong/plugins/opentelemetry/otlp.lua
Comment (Water-Melon) [reply kong/plugins/opentelemetry/otlp.lua]: updated

File: kong/db/schema/others/declarative_config.lua
Comment (Water-Melon) [reply kong/db/schema/others/declarative_config.lua]: yes

File: spec/01-unit/09-balancer/05-worker_consistency_spec.lua
Comment (Water-Melon) [reply spec/01-unit/09-balancer/05-worker_consistency_spec.lua]: this has been reverted

File: spec/01-unit/27-queue_spec.lua
Comment (chronolaw) [inline spec/01-unit/27-queue_spec.lua:3]: It may conflict with #12952

File: spec/02-integration/03-db/14-dao_spec.lua
Comment (chronolaw) [inline spec/02-integration/03-db/14-dao_spec.lua:3]: It may conflict with #12952

File: spec/02-integration/09-hybrid_mode/01-sync_spec.lua
Comment (chronolaw) [inline spec/02-integration/09-hybrid_mode/01-sync_spec.lua:8]: It may conflict with #12952

File: spec/03-plugins/15-response-transformer/01-header_transformer_spec.lua
Comment (chronolaw) [reply spec/03-plugins/15-response-transformer/01-header_transformer_spec.lua:28]: Please add this comment in the code.

File: spec/03-plugins/23-rate-limiting/06-shorthand_fields_spec.lua
Comment (chronolaw) [inline spec/03-plugins/23-rate-limiting/06-shorthand_fields_spec.lua:7]: It may conflict with https://github.com/Kong/kong/pull/12952

File: spec/03-plugins/24-response-rate-limiting/06-shorthand_fields_spec.lua
Comment (chronolaw) [inline spec/03-plugins/24-response-rate-limiting/06-shorthand_fields_spec.lua:7]: It may conflict with https://github.com/Kong/kong/pull/12952

File: spec/01-unit/27-queue_spec.lua
Comment (Water-Melon) [reply spec/01-unit/27-queue_spec.lua]: modified

File: spec/02-integration/03-db/14-dao_spec.lua
Comment (Water-Melon) [reply spec/02-integration/03-db/14-dao_spec.lua]: modified

File: spec/02-integration/09-hybrid_mode/01-sync_spec.lua
Comment (Water-Melon) [reply spec/02-integration/09-hybrid_mode/01-sync_spec.lua]: modified

File: spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua
Comment (Water-Melon) [reply spec/02-integration/09-hybrid_mode/09-config-compat_spec.lua]: modified

File: spec/03-plugins/23-rate-limiting/06-shorthand_fields_spec.lua
Comment (Water-Melon) [reply spec/03-plugins/23-rate-limiting/06-shorthand_fields_spec.lua]: modified

File: spec/03-plugins/24-response-rate-limiting/06-shorthand_fields_spec.lua
Comment (Water-Melon) [reply spec/03-plugins/24-response-rate-limiting/06-shorthand_fields_spec.lua]: modified

File: spec/03-plugins/29-acme/07-shorthand_fields_spec.lua
Comment (Water-Melon) [reply spec/03-plugins/29-acme/07-shorthand_fields_spec.lua]: modified

**Review by chronolaw — APPROVED** [review]

**Review by ADD-SP — APPROVED** [review]

**Pull request merged**
